openhands-sdk/openhands/sdk/llm/llm_registry.py

```
@@ -114,6 +120,114 @@ def add(self, llm: LLM) -> None:
+
+    def get_profile_path(self, profile_id: str) -> Path:
+        """Return the path where profile_id is stored."""
+
+        return self.profile_dir / f"{profile_id}.json"
+
+    def load_profile(self, profile_id: str) -> LLM:
+        """Load profile_id from disk and return an LLM."""
+
+        path = self.get_profile_path(profile_id)
+        if not path.exists():
+            raise FileNotFoundError(f"Profile not found: {profile_id} -> {path}")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)